activiti-cloud-services-org/activiti-cloud-services-org-service/src/main/java/org/activiti/cloud/services/organization/validation/ProcessModelValidator.java

```diff
@@ -55,29 +65,37 @@
     private final ProcessValidator processValidator;
 
     private final ProcessModelContentConverter processModelContentConverter;
+    private final ConnectorModelContentConverter connectorModelContentConverter;
 
     public final String NO_ASSIGNEE_PROBLEM_TITLE = "No assignee for user task";
     public final String NO_ASSIGNEE_DESCRIPTION = "One of the attributes 'assignee','candidateUsers' or 'candidateGroups' are mandatory on user task";
     public final String USER_TASK_VALIDATOR_NAME = "BPMN user task validator";
 
+    public final String INVALID_SERVICE_IMPLEMENTATION_PROBLEM = "Invalid service implementation";
+    public final String INVALID_SERVICE_IMPLEMENTATION_DESCRIPTION = "Invalid service implementation on service '%s'";
+    public final String SERVICE_USER_TASK_VALIDATOR_NAME = "BPMN service task validator";
+
     @Autowired
     public ProcessModelValidator(ProcessModelType processModelType,
                                  ProcessValidator processValidator,
-                                 ProcessModelContentConverter processModelContentConverter) {
+                                 ProcessModelContentConverter processModelContentConverter,
+                                 ConnectorModelContentConverter connectorModelContentConverter) {
         this.processModelType = processModelType;
         this.processValidator = processValidator;
         this.processModelContentConverter = processModelContentConverter;
+        this.connectorModelContentConverter = connectorModelContentConverter;
     }
 
     @Override
-    public void validateModelContent(byte[] bytes) {
+    public void validateModelContent(byte[] bytes,
+                                     ValidationContext validationContext) {
         try {
             BpmnModel bpmnModel = processModelContentConverter.convertToBpmnModel(bytes);
 
             List<ValidationError> validationErrors = processValidator.validate(bpmnModel);
-            List<ValidationError> validationModelAssigneeErrors = validateTasksAssignedUser(bpmnModel);
-            validationErrors.addAll(validationModelAssigneeErrors);
-
+            validationErrors.addAll(validateUserTasksAssignee(bpmnModel));
```

Comment:
You can collect all errors in Streams, concatenate the streams and collect all the errors to in the end.